vendor/brave-ios/Ads/BATBraveAds.mm

```diff
@@ -253,6 +253,43 @@ - (void)applicationDidBackground
   ads->OnBackground();
 }
 
+#pragma mark - History
+
+- (NSArray<NSDate *> *)getAdsHistoryDates
+{
+  if (![self isAdsServiceRunning]) { return @[]; }
+  const auto history = ads->GetAdsHistory(ads::AdsHistoryFilterType::kNone);
+  const auto dates = [[NSMutableArray<NSDate *> alloc] init];
+  for (auto it = history.begin(); it != history.end(); ++it) {
+    for (auto& item : it->second) {
+      for (auto& detail : item.details) {
+        const auto date = [NSDate dateWithTimeIntervalSince1970:
+                           detail.timestamp_in_seconds];
+        [dates addObject:date];
+      }
+    }
+  }
+  return dates;
+}
+
+- (BOOL)hasViewedAdsInPreviousCycle
+{
+  const auto calendar = NSCalendar.currentCalendar;
```

Comment:
See my concern about calendar standarization in other pr https://github.com/brave/brave-ios/pull/1988